evm-template/node/src/rpc/eth.rs

```diff
@@ -40,13 +40,13 @@ pub struct EthDeps<B: BlockT, C, P, A: ChainApi, CT, CIDP> {
     /// Whether to enable dev signer
     pub enable_dev_signer: bool,
     /// Network service
-    pub network: Arc<NetworkService<B, B::Hash>>,
+    pub network: Arc<dyn NetworkService>,
```

Comment:
Is there any reason to mix generics and trait objects?

Reply:
Upstream has refactored Network service, hence these changes were also introduced in upstream. I simply followed them tbh